code/3_data_processing.py

```diff
@@ -45,67 +45,84 @@
 
 import os
 import sys
+
     
 def main():
-  
-    from pyspark.sql import SparkSession
-    from pyspark.sql import functions as sqlfn
-
-    spark = (
-        SparkSession.builder.appName("Airline Data Exploration")
-        .config("spark.executor.memory", "8g")
-        .config("spark.executor.cores", "4")
-        .config("spark.driver.memory", "20g")
-        .config("spark.executor.instances", "4")
-        .config("spark.yarn.access.hadoopFileSystems", os.environ["STORAGE"])
-        .getOrCreate()
-    )
+    import cml.data_v1 as cmldata
+    from pyspark import SparkContext
+    from pyspark.sql import functions as sqlfn   
+    
+    ### cleanup needed start
+    # These shound't be set after cml.data adds iceberg support
+    import glob
+
+    jars = glob.glob("/opt/spark/optional-lib/*.jar")
+    jarsList = ",".join(jars)
+    SparkContext.setSystemProperty("spark.jars",jarsList)
+
+    SparkContext.setSystemProperty("spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog")
+    SparkContext.setSystemProperty("spark.sql.catalog.spark_catalog.type", "hive")
+    SparkContext.setSystemProperty("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+    ### cleanup needed end
+
+
+    SparkContext.setSystemProperty('spark.executor.cores', '4')
+    SparkContext.setSystemProperty('spark.executor.memory', '8g')
+
+    SPARK_CONNECTION_NAME = os.getenv("SPARK_CONNECTION_NAME")
```

Comment:
i am confused about how this is working.. how are you getting the spark connection name ?